src/function/table/system/duckdb_columns.cpp

```diff
@@ -149,20 +150,36 @@ class TableColumnHelper : public ColumnHelper {
 
 class ViewColumnHelper : public ColumnHelper {
 public:
-	explicit ViewColumnHelper(ViewCatalogEntry &entry) : entry(entry) {
+	explicit ViewColumnHelper(ViewCatalogEntry &entry, ClientContext &context) : entry(entry) {
+		auto info = entry.GetInfo();
+		auto binder = Binder::CreateBinder(context);
+		auto &view_info = info->Cast<CreateViewInfo>();
+		binder->BindCreateViewInfo(view_info);
```

Comment:
`BindCreateViewInfo` can be quite expensive - since it binds a query (which can involve I/O, e.g. if the view contains a reference to a CSV or Parquet file).

Is there any way we can avoid doing that here?

Reply:
If we only display the columns that were explicitly aliased that could work.
But otherwise there is no way to be sure that the view is up-to-date with the query it's referencing.

But maybe since this is a metadata function that might not matter?
I think in that case it might be better to just not output column data for views here then, instead of faking it

Reply:
I don't quite follow why the behavior has changed in this PR - wasn't that always the case? Shouldn't returning the `BoundNames` give the same behavior as prior to this PR?

Reply:
What we did previously was save the bound names in the `aliases` vector. Which is also where we store the explicit aliases set by the user when creating the view.

For sql generating purposes we have to distinguish between explicitly set aliases and column names taken from the state of the query at the time of creation.

I added a separate vector for that purpose, but that would break behavior between existing and newly created views, so I've removed that and instead opted for rebinding in places where the full names/types of the view are required.